sdk/tables/Azure.Data.Tables/src/TableConnectionString.cs

```diff
@@ -0,0 +1,434 @@
+        /// <param name="error">A callback for reporting errors.</param>
+        /// <returns>If true, the parse was successful. Otherwise, false.</returns>
+        internal static bool ParseInternal(string connectionString, out TableConnectionString accountInformation, Action<string> error)
+        {
+            IDictionary<string, string> settings = ParseStringIntoSettings(connectionString, error);
+
+            // malformed settings string
+
+            if (settings == null)
+            {
+                accountInformation = null;
+                return false;
+            }
+
+            // helper method
+
+            string settingOrDefault(string key)
+            {
+                settings.TryGetValue(key, out var result);
+                return result;
+            }
+
+            // devstore case
+
+            if (settings.TryGetValue(TableConstants.ConnectionStrings.UseDevelopmentSetting, out var useDevSettings) && useDevSettings == "true")
+            {
+                accountInformation =
+                    settings.TryGetValue(TableConstants.ConnectionStrings.DevelopmentProxyUriSetting, out var proxyUri)
+                    ? GetDevelopmentStorageAccount(new Uri(proxyUri))
+                    : DevelopmentStorageAccount;
+
+                return true;
+            }
+
+            // non-devstore case
+            string sasToken = default;
+
+            var matchesExplicitEndpointsSpec =
+                settings.TryGetValue(TableConstants.ConnectionStrings.TableEndpointSetting, out var tableEndpoint) &&
+                Uri.IsWellFormedUriString(tableEndpoint, UriKind.Absolute) &&
+                settings.TryGetValue(TableConstants.ConnectionStrings.SharedAccessSignatureSetting, out sasToken);
+
+            var matchesAutomaticEndpointsSpec = settings.TryGetValue(TableConstants.ConnectionStrings.AccountNameSetting, out var accountName) &&
+                settings.TryGetValue(TableConstants.ConnectionStrings.AccountKeySetting, out var accountKey) &&
+                    (settings.TryGetValue(TableConstants.ConnectionStrings.TableEndpointSetting, out accountName) ||
+                    settings.TryGetValue(TableConstants.ConnectionStrings.EndpointSuffixSetting, out var endpointSuffix));
+
+            if (matchesAutomaticEndpointsSpec || matchesExplicitEndpointsSpec)
+            {
+                if (matchesAutomaticEndpointsSpec && !settings.ContainsKey(TableConstants.ConnectionStrings.DefaultEndpointsProtocolSetting))
+                {
+                    settings.Add(TableConstants.ConnectionStrings.DefaultEndpointsProtocolSetting, "https");
+                }
+
+                var tableSecondaryEndpoint = settingOrDefault(TableConstants.ConnectionStrings.TableSecondaryEndpointSetting);
+
+                // if secondary is specified, primary must also be specified
+
+                static bool s_isValidEndpointPair(string primary, string secondary) =>
```

Comment:
nit: not sure we have official naming rules for local functions (though they seem to lean in the direction of PascalCase) but I haven't seen `s_` prefix used anywhere.